lib/components/modal.vue

```diff
@@ -69,20 +69,32 @@
         opacity: 0 !important;
     }
 
-    /* Make modal display as block instead of inline style, and because Vue's v-show deletes inline "display" style*/
+    /* Make modal display as block instead of inline style, and because Vue's v-show deletes inline "display" style */
     .modal {
         display: block;
     }
 </style>
 
 <script>
+    import generateId from '../mixins/generate-id';
     import bBtn from './button.vue';
 
+    const SELECTOR = [
+        'button:not([disabled])',
+        'input:not([disabled])',
+        'select:not([disabled])',
+        'textarea:not([disabled])',
+        'a:not([disabled]):not(.disabled)',
+        '[tabindex]:not([disabled]):not(.disabled)'
+    ].join(',');
+
     export default {
+        mixins: [generateId],
```

Comment:
`generateId` usage should be limited to inputs only as is mentioned inside Mixin. Also as of vue 2.3.x we can not trust `_uid` because they are inconsistent between client server in SSR mode. See [here](https://github.com/vuejs/vue/issues/5573) 

> _uid is internal and should not be used as part of the application state